Client/Frontend/Browser/RecentlyClosedTabs.swift

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+public protocol RecentlyClosedTabs {
+    func addTab(url: NSURL?, title: String?, faviconURL: String?)
+    func getData() -> [String:[String]]?
+    func clearData()
+    func count() -> Int
+}
+
+public class RecentlyClosedTabsStore: RecentlyClosedTabs {
+    let prefs: Prefs
+
+    public init(prefs: Prefs) {
+        self.prefs = prefs
+    }
+
+    public func addTab(url: NSURL?, title: String?, faviconURL: String?) {
+        guard var recentlyClosedTabs = prefs.dictionaryForKey("recentlyClosedTabs"),
+            var urls = recentlyClosedTabs["urls"] as? [String],
+            var titles = recentlyClosedTabs["titles"] as? [String],
+            var faviconUrls = recentlyClosedTabs["faviconUrls"] as? [String] else {
+                var recentlyClosedTabsDict = [String:[String]]()
+                recentlyClosedTabsDict["urls"] = [url?.absoluteString ?? ""]
+                recentlyClosedTabsDict["faviconUrls"] = [faviconURL ?? ""]
+                recentlyClosedTabsDict["titles"] = [title ?? ""]
```

Comment:
If we're sticking with prefs as storage, I think it'd be better to store the list as an array of dicts rather than a dict of arrays. That way, we don't have to worry about the arrays being out-of-sync.